### PR TITLE
100% Coverage web3-utills

### LINKED ISSUE #7031: Increase testing coverage - web3-utils
Goal is 100%

[Codecov](https://app.codecov.io/gh/web3/web3.js/tree/4.x/packages%2Fweb3-utils%2Fsrc)

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #7031 
Coverage lines go up to 98%, the lines that are not covered is from formatters and that needs to be revisited and fixed to work as intended 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-utils/src/formatter.ts [hunk 1]
```diff
@@ -74,11 +72,7 @@ const findSchemaByDataPath = (
 			result = (result.properties as Record<string, JsonSchema>)[dataPart];
 		} else if (result.items && (result.items as JsonSchema).properties) {
 			const node = (result.items as JsonSchema).properties as Record<string, JsonSchema>;
-
-			if (!node) {
```
Comment (luu-alex) [inline packages/web3-utils/src/formatter.ts:78]: we remove this
this conditional can never be met, 
`} else if (result.items && (result.items as JsonSchema).properties) {`
we make sure `result.properties` always exists

File: packages/web3-utils/src/formatter.ts [hunk 2]
```diff
@@ -57,10 +57,8 @@ const findSchemaByDataPath = (
 
 	for (const dataPart of dataPath) {
 		if (result.oneOf && previousDataPath) {
-			const path = oneOfPath.find(function (element: [string, number]) {
```
Comment (luu-alex) [inline packages/web3-utils/src/formatter.ts:60]: changed this for readability

File: packages/web3-utils/src/promise_helpers.ts [hunk 3]
```diff
@@ -85,25 +83,27 @@ export function pollTillDefinedAndReturnIntervalId<T>(
...
+						const res = await waitWithTimeout(func, interval);
+
+						if (!isNullish(res)) {
+							clearInterval(intervalId);
+							resolve(res as unknown as Exclude<T, undefined>);
+						}
+					} catch (error) {
 						clearInterval(intervalId);
-						resolve(res as unknown as Exclude<T, undefined>);
+						reject(error);
```
Comment (luu-alex) [inline packages/web3-utils/src/promise_helpers.ts:100]: this file was changed from prettier

**Comment by luu-alex** [comment]
Some lines were not covered as `formatter` is not working 100% as intended, need to look into the formatter function before we cover more of those lines

**Comment by luu-alex** [comment]
Looking into the format tests right now and trying to make them work

**Review by jdevcs — APPROVED** [review]

**Pull request merged**